### PR TITLE
CLDCONNECT-2448 | Migrated to DevKit 3.7.1

### PR DESCRIPTION
**Pull request by urbanusjam**

### CODE REVIEW

File: src/main/java/org/mule/modules/github/Config.java [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+import org.mule.api.annotations.components.ConnectionManagement;
+import org.mule.api.annotations.display.Password;
+import org.mule.api.annotations.param.ConnectionKey;
+import org.mule.api.annotations.param.Default;
+import org.mule.modules.github.service.ServiceFactory;
+import org.mule.util.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@ConnectionManagement(friendlyName = "Configuration")
```
Comment (cmartinezledo) [inline src/main/java/org/mule/modules/github/Config.java]: Missing configElementName

File: pom.xml [hunk 2]
```diff
@@ -1,29 +1,90 @@
...
+                <version>1.2.1</version>
+                <executions>
+                    <execution>
+                        <phase>initialize</phase>
+                        <goals>
+                            <goal>wget</goal>
+                        </goals>
+                        <configuration>
+                            <url>https://raw.githubusercontent.com/mulesoft/connector-certification-docs/develop/docs/02-coding-standards-best-practices/files/connectors-style-convention.xml</url>
+                            <outputFileName>connectors-style-convention.xml</outputFileName>
```
Comment (martinpaoloni) [inline pom.xml]: I thought @veigap didn't want this to be run in each connector build.

**Comment by martinpaoloni** [comment]
Looks good!

**Comment by cmartinezledo** [comment]
@GithubConnector:1240 You should call the client operation and not the processor

File: pom.xml [hunk 3]
```diff
@@ -1,29 +1,90 @@
...
+    </build>
+            
     <dependencies>
         <dependency>
             <groupId>org.eclipse.mylyn.github</groupId>
             <artifactId>org.eclipse.egit.github.core</artifactId>
-            <version>2.0.1</version>
+            <version>2.1.5</version>
+        </dependency>
+        <!-- fixes a classloader issue in 9.0.4 -->
```
Comment (marianogoldman) [inline pom.xml]: 9.0.4 --> 0.9.4

File: src/main/java/org/mule/modules/github/Config.java [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+
+        // Validate scope
+        if (StringUtils.isBlank(scope)) {
+            logger.warn("Empty scope");
+        }
+
+        try {
+            setServiceFactory(new ServiceFactory(connectionUser, connectionPassword, scope));
+        } catch (IOException e) {
+            logger.error(e.getMessage());
```
Comment (marianogoldman) [inline src/main/java/org/mule/modules/github/Config.java]: No need to log here as we're rethrowing with a cause.

File: src/main/java/org/mule/modules/github/Config.java [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+import org.slf4j.LoggerFactory;
+
+@ConnectionManagement(friendlyName = "Configuration")
+public class Config {
+
+    private static final Logger logger = LoggerFactory.getLogger(Config.class);
+
+    private ServiceFactory serviceFactory;
+
+    public ServiceFactory getServiceFactory() throws IOException {
```
Comment (marianogoldman) [inline src/main/java/org/mule/modules/github/Config.java]: Why throws IOException?

Comment (urbanusjam) [inline src/main/java/org/mule/modules/github/Config.java]: It is inherited from one of the GitHub API classes.

File: src/main/java/org/mule/modules/github/service/ServiceFactory.java [hunk 6]
```diff
@@ -0,0 +1,288 @@
...
+    public String getAccessToken(String scope) throws IOException {
+
+        List<String> scopes = null;
+
+        List<Authorization> authorizations = getOAuthService().getAuthorizations();
+
+        if (!authorizations.isEmpty()) {
+            if (scope != null && !scope.isEmpty()) {
+                scopes = Arrays.asList(scope.split(","));
+                if (scopes != null && !scopes.isEmpty()) {
```
Comment (marianogoldman) [inline src/main/java/org/mule/modules/github/service/ServiceFactory.java]: scopes will not be null here.

File: src/main/java/org/mule/modules/github/service/ServiceFactory.java [hunk 7]
```diff
@@ -0,0 +1,288 @@
...
+
+    public ServiceFactory(String user, String password, String scope) throws IOException {
+        this.user = user;
+        this.password = password;
+        this.token = getAccessToken(scope);
+    }
+
+    public String getAccessToken(String scope) throws IOException {
+
+        List<String> scopes = null;
```
Comment (marianogoldman) [inline src/main/java/org/mule/modules/github/service/ServiceFactory.java]: Let's declare scopes further down, closer to where it is actually initialized and used.

**Comment by cmartinezledo** [comment]
+1

**Comment by marianogoldman** [comment]
+1

**Pull request merged**